api/agent/agent.go

```diff
@@ -718,49 +740,49 @@ func (a *agent) runHot(ctx context.Context, call *call, tok ResourceToken, state
 			}
 
 			isFrozen := false
-			elapsed := time.Duration(0)
-			freezerTicker := a.freezeIdleMsecs
-			idleTimeout := time.Duration(call.IdleTimeout) * time.Second
 
 			done := make(chan struct{})
 			state.UpdateState(ctx, ContainerStateIdle, call.slots)
 			s := call.slots.queueSlot(&hotSlot{done, errC, container, nil})
 
+			stopTimer(freezeTimer)
+			stopTimer(idleTimer)
+			stopTimer(ejectTimer)
```

Comment:
since there's a select with a default i'm not sure we'll properly drain any of the timer.C channels, per the docs this will fail spectacularly if we call reset afterwards. i think as it is the select statement should pick the timer.C receive case but there may be a runtime scheduling race in doing that since if timer.C is not ready to send the default case will be selected (the docs just do `if timer.Stop() { <-timer.C }` which may wait an implied short amount of time, as i'm sure you know). idk for sure ?

Reply:
Changing this...